src/Symfony/Component/HttpKernel/Tests/KernelTest.php

```diff
@@ -626,12 +626,13 @@ public function getContainerClass(): string
     }
 
     /**
-     * Returns a mock for the BundleInterface.
+     * Returns a dummy implementation for the BundleInterface.
      */
     protected function getBundle($dir = null, $parent = null, $className = null, $bundleName = null): BundleInterface
     {
+        $bundle = $this->createStub(BundleInterface::class);
         $bundle = $this
-            ->getMockBuilder(BundleInterface::class)
+            ->getMockBuilder(\get_class($bundle))
```

Comment:
```suggestion
            ->getMockBuilder($className ?? \get_class($bundle))
```

Can't we use this and remove the `if` below?

Reply:
No, it throws:

<img width="1402" alt="Capture d’écran 2024-04-30 à 16 41 48" src="https://github.com/symfony/symfony/assets/2144837/5c09e09e-5c23-409d-b927-c1115dde163a">

However, I pushed a simpler way of creating the mock. Don't know why I didn't think about it before, but it's better now as we avoid creating 2 mocks.

Reply:
Also I never played with `setMockClassName`, but it seems that using `??` would change the semantic and the mock wouldn't implement `BundleInterface` anymore. We _need_ to implement `BundleInterface`, which is done with `getMockBuilder`. Then `setMockClassName` would rename the mock but keep `BundleInterface` as the mock parent.

Not sure I'm super clear here 😄